src/handle-data.ts

```diff
@@ -58,6 +59,34 @@ export function getTotalInfo(layout: TableLayout, colIdx: number, pageColIdx: nu
   return '';
 }
 
+/**
+ * Gets the totals alignment for head, totals and body
+ * bodyTextAlign is later used to determine independent alignment for each body cell
+ */
+export function getAlignInfo(
```

Comment:
Maybe just me but I find the logic in this function hard to read. Something like this make my head spin a bit less:

```js
export function getAlignInfo(
  textAlign: TextAlign,
  qDimensionType: EngineAPI.DimensionType | undefined,
  isDim: boolean
) {
  if (!textAlign || textAlign.auto) {
    const autoHeadTextAlign = qDimensionType === 'N' || qDimensionType === undefined ? 'right' : 'left';
    const autoTotalsTextAlign = isDim ? 'left' : 'right';

    return {
      headTextAlign: autoHeadTextAlign,
      totalsTextAlign: autoTotalsTextAlign,
      bodyTextAlign: 'auto',
    };
  }

  return {
    headTextAlign: textAlign.align,
    totalsTextAlign: textAlign.align,
    bodyTextAlign: textAlign.align,
  };
}
```

Reply:
may be even dont need  extra variables for 'autoHeadTextAlign' and 'autoTotalsTextAlign'

Reply:
oh wow we check the same thing three times. yes this is way better